appveyor.yml

```diff
@@ -1,10 +1,22 @@
-version: 1.0.{build}
-configuration: Release
-image: Visual Studio 2019
-pull_requests:
-  do_not_increment_build_number: true
```

Comment:
I feel that we should preserve most if not all existing options.
I can prepare a PR to https://github.com/nuke-build/nuke if we want to do this.

By the way, `do_not_increment_build_number` option was added in commit https://github.com/aspnetboilerplate/aspnetboilerplate/commit/4d2ed998c912681edc814421f36d38b5294e9042.
This PR also removes the `version` option, which is oddly causing build versions to revert to `1.4.3-beta{build}` from before that commit. I wonder if it's something set in AppVeyor UI.

Reply:
Oh, I just realised that @matkoch is the maintainer of NUKE.

Can we add functionality in NUKE to preserve our existing options?

Reply:
Sure I can do that. Although I thought you'd be moving to [Azure Pipelines](https://github.com/aspnetboilerplate/aspnetboilerplate/issues/5150#issuecomment-568157074)?! But anyways, I can do both!

Reply:
@matkoch for azuira pipelines build, i was planning to use vs tool to setup CI/CD instead of using build script. 

Can nuke run on appveyor only?

Reply:
It can run on any environment you like. I don't know anything about some VS tool to setup Azure Pipelines, but I think you might want to use the built-in generation of nuke. I just recently [blogged about this](https://ithrowexceptions.com/2020/02/04/yaml-hell-in-devops.html). If you want more opinions on that, I can also recommend joining the nuke slack workspace. If you find anything in your tool, which nuke doesn't support, let me know, and I will make it happen :)

Reply:
@acjh I've disabled auto-generation for appveyor.yml for now. I will include the option in the next release of nuke, and then send another pull-request.

Reply:
It's not fully reverted yet.

Reply:
Now it is. `Configuration` is determined in the build.

Reply:
1. Please remove the "auto-generated" header since it is not auto-generated and is misleading if someone just updates the file thinking that it was auto-generated before.
2. Why do we remove `"test": off`? Did we check why it was added in the first place?

Reply:
1. It would also be false to say that it's manually written. There is still the option to regenerate it, and then merge with Git.
2. https://www.appveyor.com/docs/appveyor-yml/

Reply:
> 1. It would also be false to say that it's manually written.

It was probably copy-pasted and modified. There is no header that claims it is manually written.

> There is still the option to regenerate it, and then merge with Git.

What?

> 2. https://www.appveyor.com/docs/appveyor-yml/

That does not answer the question.

Reply:
1. It's written in the header. For manual generation, execute `nuke --configure-build-server --host AppVeyor`. There was nothing copy-pasted. Most of the file is generated. Only three LOC were added manually. And as mentioned in the issue, I expected you to move to Azure Pipelines anyways. So I'm a bit confused?!
2. The docs say that `tests: off` will disable automatic execution of tests. The execution of tests is _inside_ the `Build.cs`, and there is no second test invocation. So how exactly can I further resolve your concern?

Reply:
> 1. It's written in the header. For manual generation, execute `nuke --configure-build-server --host AppVeyor`. There was nothing copy-pasted. Most of the file is generated. Only three LOC were added manually.

I mean the existing appveyor.yml was probably copy-pasted. The header is misleading in that it is generated while there are rest of the file is not.

> And as mentioned in the issue, I expected you to move to Azure Pipelines anyways. So I'm a bit confused?!

It doesn't matter what we intend to move to; we are still using AppVeyor as of this PR. Why are you confused that I am opposed to indiscretionary modification of a known-working file?

>> 2. ... Did we check why it was added in the first place?

It appears not. But ok.

Reply:
1. I removed the header. I still think of this as the less optimal choice, but it's your take :) Excluded the header, there is a total of 12 LOC, of which 3 LOC are manually added. The header clearly indicates how manual generation works. OTOH, `appveyor.yml` is not really complex regarding generation.
2. The `tests: off` is just unnecessary as far as I'm concerned. The relevant commit is 46b94be036e592c6f2e23ecdd6c94fd8a8c095b8 which introduced cake.

Reply:
Thank you.

@ryancyq Ok with the generated azure-pipelines.yml? Since it is currently unused, we can modify it later.

Reply:
For Azure Pipelines, you just need to setup a project and point it to the repository. That's all. Like I mentioned, you have a bit of execution time reduction. Please remember to modify it from the `Build.cs` ;)

Reply:
@matkoch the vs tool  i was trying to refer to was just visual studio image. The benefits of it is that when we have the proper project/solution configuration, triggering a build/test/pack via visual studio from our end will be simpler.

For example, when we are trying major version upgrade for .net core, changing the project setting will allow us to run the CI/DI without any hassle. 

I do understand that using dotnet cli does provide the same capability but it probably still worth it for us to try azure pipelines with visual studio image. 😃 

@acjh  i am not blocking on azure pipelines changes, just trying to keep this option open so that we can have experiments with it.  😄